jetty-ee10/jetty-ee10-annotations/src/main/java/org/eclipse/jetty/ee10/annotations/AnnotationParser.java

```diff
@@ -550,21 +551,23 @@ public void parse(final Set<? extends Handler> handlers, Resource r) throws Exce
         if (!r.exists())
             return;
 
-        if (FileID.isJavaArchive(r.getPath()))
-        {
-            parseJar(handlers, r);
-            return;
-        }
-
         if (r.isDirectory())
         {
             parseDir(handlers, r);
             return;
         }
-
-        if (FileID.isClassFile(r.getPath()))
+        else
```

Comment:
Why this change?

Reply:
See previous

Reply:
There is no need to change the established ordering: the change you've made just replaces `r.getPath()` with `r.getFileName()`

Reply:
It's far more subtle then that.

Path has knowledge of things like `.isFile` and `.isRegularFile` and `.isReadable`.
String (from `r.getFileName`) does not, it's just a String.

The original test `FileID.isClassFile(Path)` doesn't only check that the last Path segment text filename is a `*.class`, but also checks if the provided Path is a file, not allowing directories with the name `*.class` to pass as true.

So, for example, the following change would need to be made at a minimum to maintain backward compatibility to the original code.

``` diff
- if (FileID.isClassFile(r.getPath()))
+ if (Resoures.isReadableFile(r) && FileID.isClassFile(r.getFileName()))
```

That would need to occur in at least two places.
Once you see that code, you'll quickly understand the order change.